ec/provider.go

```diff
@@ -136,3 +68,86 @@ func Provider() *schema.Provider {
 		},
 	}
 }
+
+func newSchema() map[string]*schema.Schema {
```

Comment:
I'm not sure about extracting the schema into it's own function? It doesn't have nested lists or sets 🤔 To me it feels more readable to have it like it was. that way when you're looking at the `Provider` struct, you don't have to go look for the `newSchema` function.

Reply:
Yeah I understand that, but I'm not sure how we'd unit test this then? If you see in the unit tests, they're using `newSchema()` to make sure the schema is equal to the one in the provider, I'd rather use `newSchema` than a sloppy c/p?

Reply:
https://github.com/elastic/terraform-provider-ec/pull/118/files#diff-c75bbf3696fdeba7a90420204358c3a4R108

Reply:
Fair point 🤔